src/index.ts

```diff
@@ -0,0 +1,346 @@
+          'Content-Type': 'application/json'
+        },
+        method: 'PUT',
+        body: JSON.stringify(fetchResponseObj)
+      })
+      if (insertResponse.ok === false) {
+        const errorResponse = await response.json()
+        throw new ServerUtilError(
+          `Could not give user ${userInfo.name} access to database ${database}`,
+          errorResponse
+        )
+      }
+    }
+  }
+}
+
+export interface AdminUserInfo {
+  couchUri: string
+  name: string
+  password: string
+}
+
+export async function createAdminUser(userInfo: AdminUserInfo): Promise<void> {
+  const completeUri = `${userInfo.couchUri}/_node/_local/_config/admins/${userInfo.name}`
+  const response = await fetch(completeUri, {
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    method: 'PUT',
+    body: JSON.stringify(userInfo.password)
+  })
+  if (response.ok === false) {
+    const errorResponse = await response.json()
+    throw new ServerUtilError(
+      `Could not create user ${userInfo.name}`,
+      errorResponse
+    )
+  }
+}
+
+/**
+ * Describes a single database that should exist.
+ */
+export interface CouchSetup {
+  [dbName: string]: {
+    // Documents that should exactly match:
+    documents?: { [id: string]: object }
+
+    // Documents that we should create, unless they already exist:
+    templates?: { [id: string]: object }
+  }
+}
+
+/**
+ * Ensures that the requested databases exist in Couch.
+ */
+export async function prepareCouch(
+  couchUri: string,
+  setupInfo: CouchSetup
+): Promise<void> {
+  const connection = nano(couchUri)
+  const existingDbs = await connection.db.list()
+
+  for (const name of Object.keys(setupInfo)) {
+    const { documents = {}, templates = {} } = setupInfo[name]
+
+    // Create missing databases:
+    if (!existingDbs.includes(name)) {
+      await connection.db.create(name)
+      console.log(`Created database "${name}"`)
+    }
+    const currentDb: nano.DocumentScope<any> = connection.db.use(name)
+
+    // Update documents:
+    for (const id of Object.keys(documents)) {
+      const { _id, _rev, ...rest } = await currentDb.get(id).catch(() => ({}))
+
+      if (matchJson(documents[id], rest) === false) {
+        await currentDb.insert({ _id: id, _rev, ...documents[id] })
+        console.log(`Wrote document "${id}" in database "${name}".`)
+      }
+    }
+
+    // Create template documents:
+    for (const id of Object.keys(templates)) {
+      const { _id } = await currentDb.get(id).catch(() => ({}))
+
+      if (_id == null) {
+        await currentDb.insert({ _id: id, ...documents[id] })
+        console.log(`Wrote document "${id}" in database "${name}".`)
+      }
+    }
+  }
+}
+
+export async function dbReplication(
+  sourceHostname: string,
+  sourceUsername: string,
+  sourcePassword: string,
+  sourceDb: string,
+  destinationHostname: string,
+  destinationUsername: string,
+  destinationPassword: string,
+  destinationDb: string
+): Promise<void> {
```

Comment:
Change this function signature to `(sourceUrl: string, destinationUrl: string): Promise<void>`